src/Symfony/Component/DependencyInjection/SyntaxAware/SyntaxAwareDefinition.php

```diff
@@ -0,0 +1,173 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\DependencyInjection\SyntaxAware;
+
+use Symfony\Component\ExpressionLanguage\Expression;
+use Symfony\Component\DependencyInjection\Definition;
+use Symfony\Component\DependencyInjection\Reference;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * A definition class that is aware of shortcut syntaxes referring to services and expressions.
+ *
+ * There are two main shortcuts that are available whenever it makes sense:
+ *
+ *  A) Shortcut for referencing services:
+ *      - @logger becomes a Reference to logger
+ *      - @?logger becomes an optional Reference to logger
+ *
+ *  B) Shortcut for creating Expressions:
+ *      - @=service("foo") becomes an Expression('service("foo")')
```

Comment:
this is missing the fact that arguments containing a string starting with ``@`` need to escape it by doubling it (which makes this class violate OOP rules btw, as the behavior of ``addArgument('@Stof70')`` would change totally depending of the Definition class used to build the service